Feat/dummy logs

juskek opened the pull request:
**dummy logs table:** 
![image](https://github.com/Theodo-UK/OmniLog/assets/57725347/f0a220ec-5fd5-4027-84c5-c3942960c832)

**Loading:**
![image](https://github.com/Theodo-UK/OmniLog/assets/57725347/b2ecdfd3-1241-4faf-a32d-947683d3b1b4)

**Error:** 
![image](https://github.com/Theodo-UK/OmniLog/assets/57725347/2a2c97a3-fe45-451d-962a-970cbcba0285)



- Avoided using useContext for now since it is a client side function
  - Overkill for preserving logs between pages, can achieve the same effect with React.cache on the server side
Also avoided using tanstack react usequery since it is another client side function
  - Can use loading.tsx and error.tsx with Next.js instead

Lerri-Cofannos commented on front/src/app/LogsTable.tsx:
```diff
@@ -0,0 +1,39 @@
+
+type Props = {
+    logs: {
+        id: number;
+        datetime_utc: number;
+        input_string: string;
+        output_string: string;
+        total_tokens: number;
+    }[]
+}
```
Have you looked into using the prisma model here?

juskek replied:
good shout!

Lerri-Cofannos commented on front/src/app/loading.tsx:
```diff
@@ -0,0 +1,36 @@
+export default function Loading() {
+    return (
+        <main className="flex min-h-screen flex-col items-center justify-evenly p-24">
+            <div className="z-10 max-w-5xl w-full items-center justify-between font-mono text-sm lg:flex">
+                <p className="fixed left-0 top-0 flex w-full justify-center border-b border-gray-300 bg-gradient-to-b from-zinc-200 pb-6 pt-8 backdrop-blur-2xl dark:border-neutral-800 dark:bg-zinc-800/30 dark:from-inherit lg:static lg:w-auto  lg:rounded-xl lg:border lg:bg-gray-200 lg:p-4 lg:dark:bg-zinc-800/30">
+                    Omnilog
+                </p>
+                <div className="fixed bottom-0 left-0 flex h-48 w-full items-end justify-center bg-gradient-to-t from-white via-white dark:from-black dark:via-black lg:static lg:h-auto lg:w-auto lg:bg-none">
+                    <a
+                        className="pointer-events-none flex place-items-center gap-2 p-8 lg:pointer-events-auto lg:p-0"
+                        href="https://www.theodo.co.uk/"
+                        target="_blank"
+                        rel="noopener noreferrer"
+                    >
+                        By Theodo UK
+                    </a>
+                </div>
+            </div>
```
Could this be a Header component?

Lerri-Cofannos commented on front/src/app/page.tsx:
Shouldn't the file have the same name as the component that it is hosting?

Lerri-Cofannos commented on front/src/data/LogsData.ts:
It might not be relevant for the time being, but it would be cool to have entries which are edge cases: empty strings, long strings, negative tokens, big token counts

Lerri-Cofannos reviewed (commented):
I'm a big fan of the loading screen!f

Lerri-Cofannos commented on front/src/app/error.tsx:
```diff
@@ -0,0 +1,23 @@
+    }, [error]);
+
+    return (
+        <Header>
+            <h2>Something went wrong!</h2>
+            <button onClick={() => reset()}>Try again</button>
+        </Header>
```
Encapsulating the body in the header seems wrong... It might be better to rename it PageLayout or something more generic, don't you think? Otherwise we can have the Header as a component that we add at the top of the page, and keep the `<main>` everywhere...

CallumHemsley commented on front/src/app/page.tsx:
```diff
@@ -1,113 +1,12 @@
+import Header from "./Header";
+import LogsTable from "./LogsTable";
+
+export default async function Home() {
```
I've never seen people do `async` for React Components 🤔 I was pretty sure this would throw an error, so I tried to do the same thing via the react compiler online: 
<img width="977" alt="image" src="https://github.com/Theodo-UK/OmniLog/assets/12123687/f0f4c4fc-0b80-4cb6-aa90-1a017e568688">

juskek replied:
client side react components cant be async, but [server components](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#fetching-data-on-the-server-with-fetch) can :)

CallumHemsley replied:
ok cool okay! wasn't aware of that, much cleaner than the 'use server' thing we saw

CallumHemsley commented on front/src/data/LogsData.ts:
```diff
@@ -0,0 +1,45 @@
+import { cache } from "react";
+
+export const LogsData = {
+    getLogs: cache(async (): Promise<llm_logs[]> => {
```
I think we need to be careful with cache, when we start fetching real data

juskek replied:
yes, on second thought I don't think that is necessary - was following [this example](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#fetching-data-on-the-server-with-third-party-libraries) because I thought cache was required to make it server-side, but it is not 

will push a new commit, good catch!

Lerri-Cofannos reviewed (commented):
All good on my side!

CallumHemsley commented:
Couple of comments in general:
- The table is quite hard to read currently (I know our basic mvp was the be able to read the data, but I wonder if we can style it slightly more clearly)
- For the error screen, I'm a little bit confused about the lines going along. Also where is the actual error message going to be shown? 👀 

PS loading skeleton is sick btw!

CallumHemsley commented:
I saw you've edited the error screenshot, looking much cleaner now 🔥

juskek commented:
> Also, could we rename app to be 'home' or something? 🤔

unfortunately it's part of the [Next.js api](https://nextjs.org/docs/app/building-your-application/routing/defining-routes)

juskek commented:
> Couple of comments in general:
> 
> * The table is quite hard to read currently (I know our basic mvp was the be able to read the data, but I wonder if we can style it slightly more clearly)
> * For the error screen, I'm a little bit confused about the lines going along. Also where is the actual error message going to be shown? 👀
> 
> PS loading skeleton is sick btw!

- Updated screenshot with new table style
- Updated error screenshot

juskek commented:
> EDIT I realised what I meant here is that we currently don't have a pages directory which is the standard for next, see here: https://nextjs.org/docs/getting-started/project-structure

I think App router is the [newer version](https://nextjs.org/docs#app-router-vs-pages-router) of Pages router

CallumHemsley commented on front/src/app/LogsTable.tsx:
```diff
@@ -0,0 +1,35 @@
+import { llm_logs } from "@prisma/client";
```
I'm thinking maybe we change the name llm_logs before we fully commit - it's kinda ugly for the type to not be PascalCase
